### ISSUE TITLE
SerializerPojo and MappedFileVol issues - on 1.0 release

### ISSUE THREAD

**Issue by ZenoFuturista** [jankotek/mapdb#409]
Hello all,
I found two problems on older 1.0 release:

1) Consider this enum:
```java
public enum TestEnum {
    test("abc");
    private String value;
    
    private TestEnum(String value)
    {
        this.value = value;
    }

    public String getValue() {
        return value;
    }
}
```
Now if you take and store this structure into mapdb on lets say oracle jre, getFields(...) function in SerializerPojo will tell you that there are no fields on this class. But if you will do the same thing on android's dalvik, there will be 4 fields - this implies serialization/deserialization format incompatibility across different VMs. 
For now I just used this simple workaround:
```java
protected ObjectStreamField[] getFields(Class<?> clazz) {
....
    FastArrayList<ObjectStreamField> fieldsList = new FastArrayList<ObjectStreamField>();
    while (streamClass != null && !clazz.isEnum()) {
        for (ObjectStreamField f : streamClass.getFields()) {
...
}
```
I was not able to figure out some better fix. Personally I dont like this approach, maybe you will think out something better, so I will leave the decision on how to overcome this for you to make.

2) I am not quite sure how I came to this issue, it was probably the change of FileChannelVol to MappedFileVol used with the same underlying mapdb or maybe chunkShift was changed, anyway this does not matter. To put it simply, if MappedFileVol was instantiated on an index or phys file smaller than some multiple of chunkSize, number of chunks was incorrectly derived. In my case StoreDirect considered my index empty (chunksCount was 0) and recreated its structure, for phys, there was one chunk missing. So I made a simple fix for this:
```java
public MappedFileVol(File file, boolean readOnly, long sizeLimit, int chunkShift, int sizeIncrement) {
...
    if(fileSize>0){
        //map existing data
        int chunksCount = (int) ((fileSize>>> chunkShift));
        while(chunkSize*chunksCount < fileSize)
             chunksCount++;
        chunks = new ByteBuffer[chunksCount];
...
}
```
This solved my problems, maybe it will help you too. :)

I did a quick look on master branch and it seems that these issues apply also  (with a little difference in MappedFileVol). I hope this report will somehow help you. 
Have a nice day and hold on with this amazing project!
Zeno

**Comment by jankotek** [comment]
Second part is fixed in trunk and will be fixed in MapDB 1.0.8
I will fix first part with Enum on Android soon.

**Comment by jankotek** [comment]
Wont fix
